### PR TITLE
Create `Order.itemSubtotal` virtual field

### LINKED ISSUE #57: Extend Order model to include `itemSubtotal` virtual field
The Order sequelize model should be extended to include a virtual field called `itemSubtotal`. This field will be the sum of the prices of items in the order.

- [x] calculates sum
- [x] takes `quantity` into account
- [x] tested

### PR DESCRIPTION
**Pull request by gynnantonix**
This PR is huge and I apologize, but it _will_ make life easier on the backend and result in less/dryer code overall.  Here's what it does:
- adds the cart as a first-class concept on the User model, making it possible to get a user's cart by simply calling `.getCart()`
- speeds up test setup by having each database integration suite define its own, minimal test data set
- makes use of `cart.addProducts()` and friends
- replaces hooks methods for handling passwords (which were bloated) in favor of using a custom setter method on the User.password field
- includes test of single-user GET request (in order to exercise cart-fetching code)
- adds `Order.itemSubtotal` field and includes test of functionality

Resolves #57

### CODE REVIEW

**Review by Scutellicious — APPROVED** [review]
I approve but will not merge all of this without a second review.  This is a lot, it could have been a good learning experience if someone had helped.  In the end I'm glad it's getting done.

**Review by csweeny — APPROVED** [review]

**Pull request merged**
